### PR TITLE
terraform test setup

### PR DESCRIPTION
**Pull request by roy-sagar11**
## Summary

terraform config files to deploy bcsc-sdpr infrastructure in openshift cluster.

## Changes
- Added terraform config file to deploy a Postgres instance with persistence volume
- Added terraform config file to deploy Keycloak-x in cluster mode 
- Added terraform config file to allow external network request to reach pod through route
- Added terraform config file to allow in-cluster networking 
- Added GitHub workflow to run terraform plan on pull request
- Added GitHub workflow to run terraform apply on pull request merge against main or dev

### CODE REVIEW

File: terraform/values.yaml [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+nameOverride: ""
+
+# The number of replicas to create (has no effect if autoscaling enabled)
+replicas: 1
+
+image:
+  # The Keycloak image repository
+  repository: quay.io/keycloak/keycloak
+  # Overrides the Keycloak image tag whose default is the chart appVersion
+  tag: "latest"
```
Comment (simensma-fresh) [inline terraform/values.yaml]: Can we change this to `18.0.0`? Just want to make sure it doesn't auto-upgrade seeing as KeyCloak upgrades usually require some downtime

File: terraform/values.yaml [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+tolerations: []
+
+# Additional Pod labels
+podLabels: {}
+
+# Additional Pod annotations
+podAnnotations: {}
+
+# # Liveness probe configuration
+livenessProbe: {}
```
Comment (simensma-fresh) [inline terraform/values.yaml:177]: I'm wondering if we can reenable the liveness/readiness/startup probes now that we know the config is working :)

**Review by simensma-fresh — CHANGES_REQUESTED** [review]
Looks great! Just two small comments

**Review by iman-jamali-fw — APPROVED** [review]
Great work @roy-sagar11.

Sorry, I just started to learn Kubernetes and Terraform and cannot help much in this pr but learned a lot by reading it. Thanks

**Review by simensma-fresh — APPROVED** [review]

**Pull request merged**